packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -23,6 +44,16 @@ export interface PaymentService extends TransactionBaseService {
     data: Data
   ): Promise<PaymentSessionData>
 
+  /**
+   * @param context The type of this argument is meant to be temporary and once the previous method signature
+   * will be removed, the type will only be PaymentContext instead of Cart & PaymentContext
+   */
+  createPayment(context: Cart & PaymentContext): Promise<PaymentSessionResponse>
```

Comment:
question: why is it `Cart & PaymentContext` should this be `PaymentContext` only?

Reply:
This is to prevent any breaking changes with the actual api. At the moment a cart is expected, this is why it is still a cart plus another type mixed. Once we decide to move forward and break the api, it will be removed

Reply:
but doesn’t the signature overload ensure this? 

I don’t have experience with so just a question :)

Reply:
No it doesn't, basically what it is doing is, allowing to pass either X data or Y data, then the method knowing that both are accepted needs to handle one or the other. But it does not take i to account that the implementation of the plugin have either been updated or not. Since the plugin can be updated and use the new data or can be the old version and still rely on the cart, we have to handle the changes in two steps in order to let the consumer than the old api will be deprecated. In that case in order to have no breaking changes both are merged. Then as explained in the comment of the new signature, the Cart is temporarily added and meant to disappear so that the consumer is aware that he should not use the cart data directly and only the new data 😅 does it make sense. 

As an example, i have refactored stripe but not the other plugins, but all plugins should still works. Even the ones not refactored and still relying on the cart

Reply:
Got it makes sense!